apps/api/src/app/events/usecases/process-subscriber/process-subscriber.usecase.ts

```diff
@@ -34,11 +37,27 @@ export class ProcessSubscriber {
 
     const notification = await this.createNotification(command, template._id, subscriber);
 
-    const steps = matchMessageWithFilters(
+    let steps = matchMessageWithFilters(
       template.steps.filter((step) => step.active === true),
       command.payload
     );
 
+    const digests = steps.filter((step) => step.template.type === ChannelTypeEnum.DIGEST_BACKOFF);
+
+    if (digests.length > 0) {
+      const earliest = moment().subtract(10, 'minutes').toDate();
+      const jobs = await this.jobRepository.find({
+        updatedAt: {
```

Comment:
this will find a job for all notification templates, we need to identify a way here to associate this only to particular notification template rather than doing a generic find I guess

Reply:
By identifire right?

Reply:
Hmm So I'm not quite sure about this, from what I remember it was the identifier but it's not a unique one. Maybe worth using the notification template "_id" instead?